### DISCUSSION TITLE
Override region on AWS SDK configuration (SdkConfig struct)

### DISCUSSION THREAD

**Discussion by trevorstr** [awslabs/aws-sdk-rust#1115] [General]
I have an AWS `SdkConfig` struct that I'm sharing via an `Arc<T>`. I need to **override** the region for specific AWS SDK requests.

However, the `SdkConfig` struct doesn't have any method defined on it to specify the region that should be used. It only has a `.region()` method that takes zero parameters. This returns the region, but doesn't allow me to specify the region.

➡️ **Question**: How do I specify a region on an `SdkConfig` struct?

This seems like something that should be extremely straightforward. Shouldn't I be able to call `.region("us-west-2")` on the `SdkConfig`?

**Comment by jdisanti** [comment]
The `SdkConfig` struct is essentially immutable after it is built. If you want to override the region on it after building it, you can use its [`to_builder()` method](https://docs.rs/aws-config/latest/aws_config/struct.SdkConfig.html#method.to_builder) to convert it back to a builder, set the region, and then build it again.

**Reply by trevorstr** [reply]
Thanks, that seems to work alright.

```rust
let new_sdk = aws_creds.clone().into_builder().region(aws_config::Region::new("us-west-1")).build();
```

**Comment by ysaito1001** [comment]
> I need to override the region for specific AWS SDK requests.

If you need to override a region just for a specific operation, you can also use `config_override` on that operation.
We have [an example](https://github.com/awsdocs/aws-doc-sdk-examples/blob/5fad395aa401424b691e9905e30102c9f5afbaa7/rustv1/examples/sdk-config/src/bin/pre-request-region.rs#L52-L61) in the SDK example repository.
